src/SerializableClosure.php

```diff
@@ -159,35 +150,56 @@ public function unserialize($serialized)
     }
 
     /**
-     * Reconstruct the closure.
-     *
-     * HERE BE DRAGONS!
+     * Returns closure data for `var_dump()`.
      *
-     * The infamous `eval()` is used in this method, along with `extract()`,
-     * the error suppression operator, and variable variables (i.e., double
-     * dollar signs) to perform the unserialization work. I'm sorry, world!
+     * @return array
      */
-    private function reconstructClosure()
+    public function __debugInfo()
     {
-        // Simulate the original context the closure was created in.
-        extract($this->data['context'], EXTR_OVERWRITE);
+        return $this->data ?: $this->serializer->getData($this->closure, true);
+    }
+}
+
+/**
+ * Reconstruct a closure.
+ *
+ * HERE BE DRAGONS!
+ *
+ * The infamous `eval()` is used in this method, along with `extract()`, the
+ * error suppression operator, and variable variables (i.e., double dollar
+ * signs) to perform the unserialization work. I'm sorry, world!
+ *
+ * This is also done inside a plain function instead of a method so that the
+ * binding and scope of the closure are null.
+ *
+ * @param array $__data Unserialized closure data.
+ *
+ * @return Closure
+ * @internal
+ */
+function _reconstruct_closure(array $__data)
```

Comment:
Explained in the comment above: "This is also done inside a plain function instead of a method so that the binding and scope of the closure are null."